.github/workflows/do-not-merge-label-check.yml

```diff
@@ -0,0 +1,19 @@
+# This GitHub Action workflow is triggered on label changes for pull requests.
+# When a pull request is labeled with "DO NOT MERGE", the workflow will fails, 
+# preventing the pull request from being merged.
+# Otherwise, the workflow will succeed, allowing the pull request to be merged.
+
+name: "Do not merge check"
+on:
+  pull_request:
+    branches: [main]
+    types: [labeled, unlabeled]
+
+jobs:
+  do-not-merge-check:
+    if: ${{ github.event.label.name == 'DO NOT MERGE' && github.event.action == 'labeled' }}
```

Comment:
I'm not sure it will work like that? I mean, it looks like it will only run once when the label is added, so this will never unblock...

I see that in runtime the check is done a bit differently: https://github.com/dotnet/runtime/blob/main/.github/workflows/check-service-labels.yml

But I'm totally not an expert, so I might be wrong :)

Reply:
This works exactly as we'd like it to. I tested in a GitHub Action workflow sandbox repo that I have. It simply skips (thus implicitly succeeds) when unlabeled.

Reply:
Examples

#### Skipped

![image](https://github.com/dotnet/docs/assets/7679720/4bde7330-bb4b-449c-ab75-50e0870babb7)

#### Failed

![image](https://github.com/dotnet/docs/assets/7679720/9869c5bb-7545-406c-ae43-bcf5e74d8c31)

Reply:
just curious, I see in the examples that you touched "do not merge" label in all the tests. what if you add a different label? now I understand the YML file a bit better, so I see that it executes when any of the labels are added or removed, right? So it might "unblock" after you add any other label, despite "do not merge" still being present?

Reply:
> Skipped

If we can make it ✅ instead of 🚫, that would be nice